source/simulator/checkpoint_restart.cc

```diff
@@ -39,11 +40,25 @@ namespace aspect
     void move_file (const std::string &old_name,
                     const std::string &new_name)
     {
-      const int error = system (("mv " + old_name + " " + new_name).c_str());
+      int error = system (("mv " + old_name + " " + new_name).c_str());
 
-      AssertThrow (error == 0, ExcMessage(std::string ("Can't move files: ")
-                                          +
-                                          old_name + " -> " + new_name));
+      // If the above call failed, e.g. because there is no command-line
+      // available, try with internal functions.
+      if (error != 0)
+        {
+          if (Utilities::fexists(new_name))
+            {
+              error = remove(new_name.c_str());
+              AssertThrow (error == 0, ExcMessage(std::string ("Unable to remove file: "
+                                                               + new_name
+                                                               + ", although it seems to exist.")));
+            }
+
+          error = rename(old_name.c_str(),new_name.c_str());
```

Comment:
Ah, crap, I recognize another problem with this patch now. Previously, we used `system("mv ...")`. The `mv` command renames or moves files, between directories and if necessary from one file system to another. But the `rename` command only does the former. Among its error codes it says
```
      EXDEV  oldpath  and  newpath  are  not on the same mounted file system.  (Linux permits a file system to be mounted at multiple points, but rename() does not
              work across different mount points, even if the same file system is mounted on both.)
```
In other words, `rename` will not be able to move a file from `/tmp` to a subdirectory of `HOME` on many systems. What do we do?